src/ParameterTypeException.php

```diff
@@ -20,15 +20,17 @@ class ParameterTypeException extends ParameterAssertionException {
 	/**
 	 * @param string $parameterName
 	 * @param string $parameterType
+	 * @param mixed $parameterValue
 	 *
 	 * @throws ParameterTypeException
 	 */
-	public function __construct( $parameterName, $parameterType ) {
+	public function __construct( $parameterName, $parameterType, $parameterValue ) {
 		if ( !is_string( $parameterType ) ) {
-			throw new ParameterTypeException( 'parameterType', 'string' );
+			throw new ParameterTypeException( 'parameterType', 'string', $parameterType );
 		}
 
-		parent::__construct( $parameterName, "must be a $parameterType" );
+		$export = var_export( $parameterValue, true );
```

Comment:
Sorry, I do not like this. When the value is an array or nested object structure this produces humongous stack traces that are not usable any more. One thing you could do instead is to change the message to something like `"must be a $parameterType, got $expectedType instead"`.